kratos/python_scripts/compare_two_files_check_process.py

```diff
@@ -139,52 +152,54 @@ def __CompareResultsBlock(self, lines1, lines2, current_index):
 
         return current_index+2 # directly incrementing to get the new result label
 
+    def __CompareDatFile(self):
```

Comment:
This function follows a similar policy as the previous with some differences in low level detail. Step 1 - compare header lines for equality, Step 2 - compare floating point numbers with almost equal. The code would be simpler to read and probably easier to maintain if you extract the low level operations to separate functions. When reading the __Compare*File() functions one should quickly read the higher level stages without getting slowed down by the low level details.